src/components/Map/MapComponent.tsx

```diff
@@ -81,26 +102,43 @@ const MapComponent = () => {
             w="100vw"
             justifyContent="center"
           >
-            <HStack spacing={3}>
-              <Button
-                variant="solid"
-                colorScheme="red"
-                size="lg"
-                boxShadow="lg"
-                onClick={() => flyToLocation(currPosition)}
-              >
-                Curr Location
-              </Button>
-              <Button
-                variant="solid"
-                colorScheme="red"
-                size="lg"
-                boxShadow="lg"
-                onClick={() => flyToLocation([1.304833, 103.831833])}
-              >
-                Random Location
-              </Button>
-            </HStack>
+            <VStack spacing={3}>
+              <InputGroup>
+                <InputLeftElement pointerEvents='none'>
+                  <Search2Icon/>
+                </InputLeftElement>
+                <Input 
+                  placeholder="Search location..."
+                  variant="filled"
+                  onChange={
```

Comment:
we should abstract input related code out to a separate component

also just note that we want things to be mobile friendly so we need another event other than on key down -> recommed using the search icon as an onclick